airflow/providers/microsoft/teams/hooks/teams_webhook.py

```diff
@@ -0,0 +1,132 @@
+    :param subtitle: The subtitle of the message to send
+    :param action_button_name: The name of the action button
+    :param action_button_url: The URL for the action button clicked
+    :param theme_color: Hex code of the card theme, without the #
+    :param icon_url: The icon activityImage URL string to be added to message card.
+    :param proxy: Proxy to use to make the MS Teams webhook call
+    """
+
+    def __init__(
+        self,
+        http_conn_id=None,
+        webhook_token=None,
+        message="",
+        subtitle="",
+        theme_color="",
+        facts=None,
+        action_button_name="",
+        action_button_url="",
+        icon_url=None,
+        proxy=None,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(http_conn_id=http_conn_id, *args, **kwargs)
+        self.webhook_token = self._get_token(webhook_token, http_conn_id)
+        self.message = message
+        self.subtitle = subtitle
+        self.theme_color = theme_color
+        self.facts = facts
+        self.action_button_name = action_button_name
+        self.action_button_url = action_button_url
+        self.icon_url = icon_url
+        self.proxy = proxy
+
+    def _get_token(self, token: str, http_conn_id: Optional[str]) -> str:
+        """
+        Given either a manually set token or a conn_id, return the webhook_token to use.
+
+        :param token: The manually provided token
+        :param http_conn_id: The conn_id provided
+        :return: webhook_token to use
+        :rtype: str
+        """
+        if token:
+            return token
+        elif http_conn_id:
+            conn = self.get_connection(http_conn_id)
+            extra = conn.extra_dejson
+            web_token = extra.get('webhook_token', '')
+            return web_token
+        else:
+            raise AirflowException('Cannot get token: No valid Teams webhook token nor conn_id supplied')
+
+    def _build_teams_message(self) -> str:
+        """
+        Construct Teams connector message. All the relevant parameters are combined to a valid
+        Teams json message.
+
+        :return: JSON formated MS Teams connector message to send
+        :rtype: str
+        """
+        card = {
+            "themeColor": self.theme_color,
+            "summary": self.message,
+            "sections": [{
+                "activityTitle": self.message,
+                "activitySubtitle": self.subtitle,
+                "activityImage": self.icon_url,
+                "facts": self.facts,
+                "potentialAction": [{
+                    "@context": "http://schema.org",
+                    "@type": "OpenUri",
+                    "name": self.action_button_name,
+                    "targets": [{
+                        "os": "default",
+                        "uri": self.action_button_url
+                    }]
+                }]}
+            ]
+        }
+        return json.dumps(card)
+
+    def execute(self) -> None:
```

Comment:
```suggestion
    def send(self) -> None:
```

execute is fine but it may cause confusion with `execute` on Operator classes.
Lets avoid this potential confusion

Reply:
> I think there is no need for the teams_ prefix in the file names

Just like slack, we might be adding support for integrating Teams API asides webhook in the future. To maintain naming conventions and avoid filenaming issues in the future, I guess it might be best for this module to be left as teams_webhook.py and the upcoming teams API can then be named teams.py, just like slack providers naming conventions.

Reply:
OK so lets just change the `execute` to `send`